k2/python/k2/fsa_algo.py

```diff
@@ -36,6 +37,38 @@ def linear_fsa(symbols: Union[List[int], List[List[int]]]) -> Fsa:
     return fsa
 
 
+def linear_fst(symbols: Union[List[int], List[List[int]]],
+               aux_symbols: Union[List[int], List[List[int]]]) -> Fsa:
+    '''Construct an linear FST from symbols and it's corresponding
+       auxiliary symbols.
+
+    Note:
+      The scores of arcs in the returned FST are all 0.
+
+    Args:
+      symbols:
+        A list of integers or a list of list of integers.
+      aux_symbols:
+        A list of integers or a list of list of integers.
+
+    Returns:
+      An FST if the input is a list of integers.
+      A vector of FSTs if the input is a list of list of integers.
+    '''
+    ragged_arc = _k2.linear_fsa(symbols)
+    if isinstance(symbols[0], List):
+        assert isinstance(aux_symbols[0],
+                          List), 'aux_symbols and symbols does not match.'
+        for sym in aux_symbols:
+            sym.append(-1)  # -1 == kFinalSymbol
+        aux_labels = torch.IntTensor(aux_symbols).reshape(-1,)
```

Comment:
I wonder the behavior here: what's the size of `aux_labels`? it equals to `len(sym)` before appending or after appending? 

BTW, I guess it may be better if we also append -1 to aux_labels as well, so that when we do `invert` on the resulted linear_fst, everything would be fine naturally.

Reply:
The size of aux_labels is `len(sym)` after appending, because we should keep the size of aux_labels and arcs the same.
I appened -1 to aux_symbols and then converted it to torch tensor as `aux_labels`